### ISSUE TITLE
Deleted script pubkey can still be queried after deletion

### ISSUE THREAD

**Issue by MaxFangX** [bitcoindevkit/bdk#829]
## Report

**Describe the bug**
- After deleting a `script_pubkey` by path, it can still be queried by script.
- After deleting a `script_pubkey` by script, it can still be queried by path.

**To Reproduce**  

```rust
/// Reproduces some bug's in BDK's [`MemoryDatabase`].
#[test]
fn bdk_bugs() {
    use bitcoin::hashes::hex::FromHex;

    // First bug
    {
        let mut db = MemoryDatabase::new();

        let script = Script::from(
            Vec::<u8>::from_hex(
                "76a91402306a7c23f3e8010de41e9e591348bb83f11daa88ac",
            )
            .unwrap(),
        );
        let keychain = KeychainKind::External;
        let child = 0;

        db.set_script_pubkey(&script, keychain, child).unwrap();
        let del_script = db
            .del_script_pubkey_from_path(keychain, child)
            .unwrap()
            .unwrap();
        assert_eq!(del_script, script);

        // We deleted by path, but it still exists when we query by script
        assert!(db.get_path_from_script_pubkey(&script).unwrap().is_none());
    }

    // Second bug
    {
        let mut db = MemoryDatabase::new();

        let script = Script::from(
            Vec::<u8>::from_hex(
                "76a91402306a7c23f3e8010de41e9e591348bb83f11daa88ac",
            )
            .unwrap(),
        );
        let keychain = KeychainKind::External;
        let child = 0;

        db.set_script_pubkey(&script, keychain, child).unwrap();
        let (del_chain, del_child) =
            db.del_path_from_script_pubkey(&script).unwrap().unwrap();
        assert_eq!(del_chain, keychain);
        assert_eq!(del_child, child);

        // We deleted by script, but it still exists when queried by path
        assert!(db
            .get_script_pubkey_from_path(keychain, child)
            .unwrap()
            .is_none());
    }
}
```

**Expected behavior**  
After calling `del_script_pubkey_from_*`, a subsequent call to `get_script_pubkey_from_*` should return None.

**Build environment**  
 - BDK tag/commit: 0.25
 - OS+version: macOS 
 - Rust/Cargo version: 12.6

## Comments

If the above behavior is actually intended, it should be better documented, and included in the database test suite.

**Comment by notmandatory** [comment]
Hi, thank for reporting this. Right now the team is working on redoing how scripts are sync'd and stored, this is the `bdk_core` project leading up to a 1.0 stable release. If this issue isn't causing any issues for users then it's best to wait for the `bdk` 1.0 release and verify the issue is fixed then.

**Comment by notmandatory** [comment]
Need to confirm this is fixed with the new 1.0 model.

**Comment by LLFourn** [comment]
This problem doesn't apply anymore. You can't delete things :P.
